tests/test_conflicting_formatters.py

```diff
@@ -0,0 +1,72 @@
+import tempfile
+from typing import Generator, List, Type
+
+import pytest
+from _pytest.fixtures import SubRequest
+
+from pydocstringformatter import _testutils as test_utils
+from pydocstringformatter._utils.exceptions import UnstableResultError
+from pydocstringformatter.run import _Run
+
+
+@pytest.fixture(scope="function")
+def patched_run(request: SubRequest) -> Generator[Type[_Run], None, None]:
+    """Patches the _Run class to use the MakeAFormatter and MakeBFormatter."""
+    # This monkeypatches the formatters to make them conflict.
+    # and the only formatters applied
+    # pylint: disable=wrong-import-position, import-outside-toplevel
+    from pydocstringformatter import _formatting  # noqa: E402
+
+    old_formatters = _formatting.FORMATTERS
+    _formatting.FORMATTERS = request.param
+
+    def cleanup() -> None:
+        """Returns the formatters to their original state."""
+        _formatting.FORMATTERS = old_formatters
+
+    request.addfinalizer(cleanup)
+
+    yield _Run
+
+
+@pytest.mark.parametrize(
+    "patched_run,expect_errors",
+    [
+        (
+            [test_utils.MakeAFormatter(), test_utils.MakeBFormatter()],
+            ["Conflicting Formatters"],
+        ),
+        (
+            [test_utils.MakeBFormatter(), test_utils.AddBFormatter()],
+            ["not able to make stable changes"],
+        ),
+        (
+            [
+                test_utils.MakeBFormatter(),
+                test_utils.AddBFormatter(),
+                test_utils.MakeAFormatter(),
+            ],
+            ["Conflicting formatters:"],
+        ),
+    ],
+    indirect=["patched_run"],
+)
+def test_conflicting_formatters(
+    patched_run: Type[_Run],  # pylint: disable=redefined-outer-name
+    expect_errors: List[str],
+) -> None:
+    """Tests that conflicting formatters raise an error."""
+
+    with tempfile.TemporaryDirectory() as tmp:
```

Comment:
You can use the `tmp_path` `fixture` from `pytest`. We use it in other tests as well.

Reply:
Done